### DISCUSSION TITLE
Logging in Python

### DISCUSSION THREAD

**Discussion by techtanic** [vercel/community#3197] [Help] [answered]
### Summary

the print function doesnt work.
How do i write to logs

### Steps to Reproduce

print("foobbar")

**Comment by amyegan** [comment, marked as answer]
Hi @techtanic. If you us `print()` in a Python serverless function, it will be written to the [runtime logs](https://vercel.com/docs/concepts/observability/runtime-logs#).  For example, I just added `print("Hello World!)` to my Python test site and you can see it appear in the logs here

<img width="1512" alt="Python app runtime logs showing Hello World on the api route" src="https://github.com/vercel/community/assets/8247105/ff745b18-898e-4e17-a339-03d092874648">

**Comment by l3cs0** [comment]
@amyegan but that would mean that i have to deploy right ? Is there a way to see the prints while running locally

**Comment by vasanth-asokan** [comment]
@l3cs0 while running locally, you should see it on your local terminal where the server process is running. 

@amyegan -- I have Python logging based logger() statements being output to console from my server process but don't see them in the request response logs within runtime logs. Any ideas on how to find out what's going wrong, or is there some configuration needed to enable this feature?

**Comment by justsee** [comment]
@vasanth-asokan if you're using `logging.basicConfig` then logging statements don't appear to output to the Vercel runtime logs.

Instead you can configure a `StreamHandler` so logs are directed to the console, and they will then appear in the Vercel runtime logs:

```
log_level = 'INFO'
logger = logging.getLogger()
logger.setLevel(log_level)

console_handler = logging.StreamHandler()
console_handler.setLevel(log_level)
logger.addHandler(console_handler)
```
